src/vos/vos_iterator.c

```diff
@@ -226,11 +257,14 @@ vos_iter_prepare(vos_iter_type_t type, vos_iter_param_t *param,
 	iter->it_ref_cnt	= 1;
 	iter->it_parent		= NULL;
 	iter->it_from_parent	= 0;
+	iter->it_ts_set		= ts_set;
 
 	*ih = vos_iter2hdl(iter);
-
 out:
-	vos_dth_set(saved);
+	if (rc == -DER_NONEXIST && dtx_is_valid_handle(dth))
+		vos_ts_set_update(ts_set, dth->dth_epoch);
+	if (rc != 0)
+		vos_ts_set_free(ts_set);
```

Comment:
It won't ever get set in this case or returned to the user.  If prepare fails, goto out will skip setting iter->it_ts_set and otherwise, it won't be valid memory to reset it.